lua/nvim-tree/actions/fs/create-file.lua

```diff
@@ -6,12 +6,15 @@ local notify = require "nvim-tree.notify"
 
 local find_file = require("nvim-tree.actions.finders.find-file").fn
 
-local M = {}
+local M = {
+  config = {},
+}
 
 local function create_and_notify(file)
   local ok, fd = pcall(vim.loop.fs_open, file, "w", 420)
   if not ok then
-    notify.error("Couldn't create file " .. file)
+    local notify_file = M.config.notify.absolute_path and file or utils.get_last_path_elem(file)
```

Comment:
We are doing this a lot; could we extract a function to do this? `utils.lua` would be a good place.

Reply:
Yea absolutely. If I understand correctly though, the function should take in a _boolean_ and an absolute path, and return the last element or the path itself based on the boolean value?

Reply:
> the function should take in a _boolean_ and an absolute path

Ah yes, you are right. That's not very workable.

How about we put it in notify itself? That knows about `opts.notify`. The function could then just take the path and maybe strip.

Reply:
Hmm, I see, maybe be can do something like this then? 
```lua
for _, x in ipairs(modes) do
  M[x.name] = function(msg, src, dest, ...)
    local args = { ... }
    local parse_path = function(path)
      return config.absolute_path and path or vim.fn.fnamemodify(path, ":t")
    end
    if src ~= nil and dest ~= nil then
      msg = string.format(msg, parse_path(src), parse_path(dest), unpack(args))
    elseif src ~= nil then
      msg = string.format(msg, parse_path(src), unpack(args))
    elseif dest ~= nil then
      msg = string.format(msg, parse_path(dest), unpack(args))
    else
      msg = string.format(msg, unpack(args))
    end
    return dispatch(x.level, msg)
  end
end

``` 
**New Params**
`src`  -> some element path 
`dest` -> another element path (primarily to be used with _rename_ action) 
`{...}` -> other arguments to be used to format the `msg` 

I believe this could potentially be used for various different use cases.
* Notifications with only source element but other format args e.g. `notify.error("Could not create %s, because %s", node.absolute_path, nil, some_err)`
* Only destination and other format args e.g.  `notify.error("Destination folder %s doesn't exist, try %s", nil, some_path, some_advice)` (Although probably not the most _useful_ use case) 
*  With both source and destination and other args e.g. `notify.error("Could not move %s into %s, because %s", node.absolute_path, to, some_err)`
* Just source element e.g. `notify.error("%s was properly created", node.absolute_path)`
* Just other args e.g. `notify.error("Something went wrong... %s", nil, nil, some_err)`
* And other use cases. 

But most importantly I think this won't be a breaking change, because you would still be able to pass in only a `msg` and have the same functionality as before?

Reply:
I might be missing something, but that's probably a bit complex.

I was thinking something like this, that open-file and friends could call and pass to `notify.info`.

```lua
function M.render_path(path)
  if M.config.notify.absolute_path then
    return fnamemodify ...
  else
    return path
  end
end
```

Reply:
Ah! okay, yea I was definitely over complicating things here. If I understand correctly though, have `render_path` be a method of _notify_ module, and have the fs actions call upon `notify.render_path(some_path)` to retrieve the appropriate path, and then be able to use it in `notify.some_mode`?

Reply:
Yes please. Simplest/most readable solution.

Reply:
I'm a fan of this solution and it works great! However, I've made an error when squashing my commits (I had way too many unnecessary commits), so I will close this PR, and raise a cleaner one if that's okay? Sorry about that.